### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'pop'

### DISCUSSION THREAD

**Discussion by 3553531421** [zai-org/ChatGLM3#1191] [Q&A]
我使用ptuning_v2进行微调出现这个错误，我把数据集的样本弄得只用几个，因为只是想测试能不能微调成功
![image](https://github.com/THUDM/ChatGLM3/assets/139878246/daec2ded-b8f7-4a39-afa2-6fab64b06caf)

**Comment by zRzRzRzRzRzRzR** [comment]
具体报错命令是

感觉是你数据没有读入进去呢

**Comment by 3553531421** [comment]
我的验证集确实没有读进去，我现在放弃了验证集，遇到了新问题，在单机多卡的时候，使用给出的命令
OMP_NUM_THREADS=1 torchrun --standalone --nnodes=1 --nproc_per_node=8  finetune_hf.py  data/AdvertiseGen/  THUDM/chatglm3-6b  configs/lora.yaml configs/ds_zero_2.json   ，路径我已经换成自己的了，configs/ds_zero_2.json会被当成第四个参数，传给保存点，然后出现问题







------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年5月2日(星期四) 下午5:38
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [THUDM/ChatGLM3] AttributeError: &#39;NoneType&#39; object has no attribute &#39;pop&#39; (Discussion #1191)














 
具体报错命令是
 
感觉是你数据没有读入进去呢

**Comment by techmanta2026** [comment]
这个问题你解决了吗？我这边也出现了这个问题，我的数据集里面包含json数据，如果不包含就不会报这个错误。
